src/main/java/uk/ac/ic/wlgitbridge/util/Tar.java

```diff
@@ -31,33 +36,40 @@ public static InputStream zip(
                 File fileOrDir,
                 long[] sizePtr
         ) throws IOException {
-            ByteArrayOutputStream target = new ByteArrayOutputStream();
+            File tmp = File.createTempFile(fileOrDir.getName(), ".tar.bz2");
+            tmp.deleteOnExit();
+            OutputStream target = new FileOutputStream(tmp);
+            /* Closes target */
             try (OutputStream bzip2 = new BZip2CompressorOutputStream(target)) {
                 tarTo(fileOrDir, bzip2);
             }
             if (sizePtr != null) {
-                sizePtr[0] = target.size();
+                sizePtr[0] = tmp.length();
             }
-            return target.toInputStream();
+            return new FileInputStream(tmp);
         }
 
         public static void unzip(
                 InputStream tarbz2,
                 File parentDir
         ) throws IOException {
-            try (InputStream tar = new BZip2CompressorInputStream(tarbz2)) {
-                untar(tar, parentDir);
-            }
+            /* BZip2CompressorInputStream does not need closing
+               Closing it would close tarbz2 which we should not do */
+            InputStream tar = new BZip2CompressorInputStream(tarbz2);
+            untar(tar, parentDir);
         }
 
     }
 
     private Tar() {}
 
     public static InputStream tar(File fileOrDir) throws IOException {
```

Comment:
Indeed we aren't, but the code it calls is also called by the tar/bz2 stuff. So it's more like a helper function for the tests to just test the tar functionality.